task-maker-exec/src/executor.rs

```diff
@@ -261,11 +263,12 @@ impl Executor {
     #[allow(clippy::unnecessary_wraps)]
     fn handle_scheduler_messages(
         receiver: Receiver<SchedulerExecutorMessage>,
-        clients: Arc<CHashMap<ClientUuid, ChannelSender<ExecutorServerMessage>>>,
+        clients: Arc<Mutex<HashMap<ClientUuid, ChannelSender<ExecutorServerMessage>>>>,
     ) -> Result<(), Error> {
         let mut ready_files: HashMap<ClientUuid, Vec<(FileUuid, FileStoreHandle, bool)>> =
             HashMap::new();
         while let Ok((client_uuid, message)) = receiver.recv() {
+            let clients = clients.lock().unwrap();
```

Comment:
I guess this lock() won't be a big deal, especially since this Mutex is not really contented.